src/main/java/bg/nbu/gradebook/domain/entities/Class.java

```diff
@@ -0,0 +1,31 @@
+package bg.nbu.gradebook.domain.entities;
+
+import static java.util.Collections.emptySet;
+import static javax.persistence.CascadeType.ALL;
+import static javax.persistence.FetchType.EAGER;
+
+import java.util.Set;
+
+import javax.persistence.Entity;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToMany;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Entity
+@Getter
+@Setter
+@NoArgsConstructor
+@Table(name = "classes")
+public class Class extends BaseEntity {
```

Comment:
I think the Class should also have in what year they are or graduation year/starting year + some id to differentiate between different classes from the same year. Like we have 'a', 'b', ...